storage/ksql/src/main/java/io/apicurio/registry/storage/impl/ksql/KafkaSqlRegistryStorage.java

```diff
@@ -0,0 +1,518 @@
+import io.apicurio.registry.metrics.PersistenceExceptionLivenessApply;
+import io.apicurio.registry.metrics.PersistenceTimeoutReadinessApply;
+import io.apicurio.registry.storage.ArtifactAlreadyExistsException;
+import io.apicurio.registry.storage.ArtifactMetaDataDto;
+import io.apicurio.registry.storage.ArtifactNotFoundException;
+import io.apicurio.registry.storage.EditableArtifactMetaDataDto;
+import io.apicurio.registry.storage.RegistryStorageException;
+import io.apicurio.registry.storage.RuleAlreadyExistsException;
+import io.apicurio.registry.storage.RuleConfigurationDto;
+import io.apicurio.registry.storage.RuleNotFoundException;
+import io.apicurio.registry.storage.VersionNotFoundException;
+import io.apicurio.registry.storage.impl.sql.AbstractSqlRegistryStorage;
+import io.apicurio.registry.types.ArtifactState;
+import io.apicurio.registry.types.ArtifactType;
+import io.apicurio.registry.types.RegistryException;
+import io.apicurio.registry.types.RuleType;
+
+/**
+ * An implementation of a registry storage that extends the basic SQL storage but federates 'write' operations
+ * to other nodes in a cluster using a Kafka topic.  As a result, all reads are performed locally but all 
+ * writes are published to a topic for consumption by all nodes.
+ * 
+ * @author eric.wittmann@gmail.com
+ */
+@ApplicationScoped
+@PersistenceExceptionLivenessApply
+@PersistenceTimeoutReadinessApply
+@Counted(name = STORAGE_OPERATION_COUNT, description = STORAGE_OPERATION_COUNT_DESC, tags = {"group=" + STORAGE_GROUP_TAG, "metric=" + STORAGE_OPERATION_COUNT})
+@ConcurrentGauge(name = STORAGE_CONCURRENT_OPERATION_COUNT, description = STORAGE_CONCURRENT_OPERATION_COUNT_DESC, tags = {"group=" + STORAGE_GROUP_TAG, "metric=" + STORAGE_CONCURRENT_OPERATION_COUNT})
+@Timed(name = STORAGE_OPERATION_TIME, description = STORAGE_OPERATION_TIME_DESC, tags = {"group=" + STORAGE_GROUP_TAG, "metric=" + STORAGE_OPERATION_TIME}, unit = MILLISECONDS)
+@Logged
+@SuppressWarnings("unchecked")
+public class KafkaSqlRegistryStorage extends AbstractSqlRegistryStorage {
+
+    private static final Logger log = LoggerFactory.getLogger(KafkaSqlRegistryStorage.class);
+
+    @Inject
+    @ConfigProperty(name = "registry.ksql.globalRuleKey", defaultValue = "__global_rule")
+    String globalRuleKey;
+    
+    @Inject
+    @ConfigProperty(name = "registry.ksql.bootstrap.servers")
+    String bootstrapServers;
+
+    @Inject
+    @ConfigProperty(name = "registry.ksql.topic", defaultValue = "ksql-journal")
+    String topic;
+
+    @Inject
+    @ConfigProperty(name = "registry.ksql.consumer.startupLag", defaultValue = "1000")
+    Integer startupLag;
+
+    @Inject
+    @ConfigProperty(name = "registry.ksql.consumer.poll.timeout", defaultValue = "1000")
+    Integer pollTimeout;
+    
+    @Inject
+    KafkaSqlCoordinator coordinator;
+
+    @Inject
+    KafkaSqlDispatcher dispatcher;
+    
+    private boolean stopped = true;
+    private KafkaProducer<String, JournalRecord> producer;
```

Comment:
Why not using producer/consumer from Registry's Kafka Util ?

Reply:
Will do, thanks.